crates/module-system/module-implementations/sov-evm/src/lib.rs

```diff
@@ -153,6 +156,29 @@ pub struct Evm<S: Spec> {
     phantom: core::marker::PhantomData<S>,
 }
 
+/// The top-level error type for all EVM module operations.
+///
+/// This enum wraps all specific error types that can occur during different
+/// EVM operations, providing a unified error interface for the module.
+#[derive(Debug, Error, Serialize)]
+pub enum Error {
+    /// An error occurred in a core module operation.
+    #[error(transparent)]
+    CoreModuleError(#[from] CoreModuleError),
+}
+
+impl ErrorDetail for Error {
+    fn error_detail(&self) -> Result<ErrorContext, Box<dyn std::error::Error + Send + Sync>> {
+        Ok(err_detail!(self))
```

Comment:
fyi, I'm considering updating this implementation in the bank to be something like:

```
let context = err_detail!(self);
let message = self.to_string();
Ok(err_detail!{"message": message, "context" context})
```

So we get the human readable part of `thiserror` as well